aviary/subsystems/propulsion/propulsion_premission.py

```diff
@@ -80,77 +79,109 @@ def configure(self):
         if verbosity > Verbosity.VERBOSE:
             out_stream = sys.stdout
 
-        comp_list = [
-            self._get_subsystem(group)
-            for group in dir(self)
-            if self._get_subsystem(group)
-            and group not in ['pre_mission_mux', 'propulsion_sum']
-        ]
+        # Patterns to identify which inputs/outputs are vectorized and need to be
+        # split then re-muxed
+        pattern = ['engine:', 'nacelle:']
 
         # Dictionary of all unique inputs/outputs from all new components, keys are
         # units for each var
         unique_outputs = {}
-        unique_inputs = {}
+        # unique_inputs = {}
 
-        # dictionaries of inputs/outputs for each added component in prop pre-mission
+        # dictionaries of inputs/outputs for engine in prop pre-mission
         input_dict = {}
         output_dict = {}
 
-        for idx, comp in enumerate(comp_list):
+        for idx, engine_model in enumerate(engine_models):
+            engine = self._get_subsystem(engine_model.name)
             # Patterns to identify which inputs/outputs are vectorized and need to be
             # split then re-muxed
             pattern = ['engine:', 'nacelle:']
 
             # pull out all inputs (in dict format) in component
-            comp_inputs = comp.list_inputs(
-                return_format='dict', units=True, out_stream=out_stream, all_procs=True
+            eng_inputs = engine.list_inputs(
+                return_format='dict',
+                units=True,
+                out_stream=out_stream,
+                all_procs=True,
+            )
+            # switch dictionary keys to promoted name rather than full path
+            # only handle variables that were top-level promoted inside engine model
+            eng_inputs = dict(
+                [
+                    (eng_inputs[key]['prom_name'], eng_inputs[key])
+                    for key in eng_inputs
+                    if '.' not in eng_inputs[key]['prom_name']
+                ]
             )
             # only keep inputs if they contain the pattern
-            input_dict[comp.name] = dict(
-                (key, comp_inputs[key])
-                for key in comp_inputs
+            input_dict[engine.name] = dict(
+                (key, eng_inputs[key])
+                for key in eng_inputs
                 if any([x in key for x in pattern])
             )
             # Track list of ALL inputs present in prop pre-mission in a "flat" dict.
             # Repeating inputs will just override what's already in the dict - we don't
             # care if units get overridden, if they differ openMDAO will convert
             # (if they aren't compatible, then a component specified the wrong units and
             # needs to be fixed there)
-            unique_inputs.update([(key, input_dict[comp.name][key]['units'])
-                                 for key in input_dict[comp.name]])
+            # unique_inputs.update(
```

Comment:
Not sure what is going on with this being commented out.

Reply:
I went through 2 or 3 full revisions of this code so as I best remember it I used to track a list of inputs for the entire propulsion system without duplicates, but at some point was able to eliminate the need for them. I left this code in in case I actually did end up needing that list again, which looks pretty unlikely at this point. I will remove this code now